src/main/java/io/split/android/client/service/executor/SplitTaskFactoryImpl.java

```diff
@@ -60,33 +62,43 @@ public SplitTask createImpressionsRecorderTask() {
     }
 
     @Override
-    public SplitTask createSplitsSyncTask() {
+    public SplitsSyncTask createSplitsSyncTask() {
         return new SplitsSyncTask(
                 mSplitApiFacade.getSplitFetcher(),
                 mSplitsStorageContainer.getSplitsStorage(),
                 new SplitChangeProcessor());
     }
 
     @Override
-    public SplitTask createMySegmentsSyncTask() {
+    public MySegmentsSyncTask createMySegmentsSyncTask() {
         return new MySegmentsSyncTask(
                 mSplitApiFacade.getMySegmentsFetcher(),
                 mSplitsStorageContainer.getMySegmentsStorage());
     }
 
     @Override
-    public SplitTask createLoadMySegmentsTask() {
+    public LoadMySegmentsTask createLoadMySegmentsTask() {
         return new LoadMySegmentsTask(mSplitsStorageContainer.getMySegmentsStorage());
     }
 
     @Override
-    public SplitTask createLoadSplitsTask() {
+    public LoadSplitsTask createLoadSplitsTask() {
         return new LoadSplitsTask(mSplitsStorageContainer.getSplitsStorage());
     }
 
     @Override
-    public SplitTask createSseAuthenticationTask() {
+    public SseAuthenticationTask createSseAuthenticationTask() {
         return new SseAuthenticationTask(mSplitApiFacade.getSseAuthenticationFetcher(),
                 mApiKey, mUserKey, new SseChannelsParser());
     }
+
+    @Override
+    public SplitKillTask createSplitKillTask() {
+        return new SplitKillTask(mSplitsStorageContainer.getSplitsStorage());
+    }
+
+    @Override
+    public MySegmentsUpdateTask createMySegmentsUpdateTask() {
+        return new MySegmentsUpdateTask(mSplitsStorageContainer.getMySegmentsStorage());
+    }
```

Comment:
Since all usages of these tasks consist of creating it, and then setting the appropriate paramter (killed split or list of segments), would it make sense to pass that parameter at construction time? This would allow for immutable/final properties, and remove the need to call the setter afterward

Reply:
Good idea. It's done! Thanks.